mitiq/shadows/classical_postprocessing.py

```diff
@@ -0,0 +1,136 @@
+# Copyright (C) Unitary Fund
+#
+# This source code is licensed under the GPL license (v3) found in the
+# LICENSE file in the root directory of this source tree.
+"""Classical post-processing process of classical shadows."""
+
+from typing import Tuple, List, Any
+
+import cirq
+import numpy as np
+from numpy.typing import NDArray
+
+# local unitary that applied to the qubits
+phase_z = cirq.S._unitary_().conj()
+hadamard = cirq.H._unitary_()
+identity = cirq.I._unitary_()
+PAULI_MAP = {"X": hadamard, "Y": hadamard @ phase_z, "Z": identity}
+
+
+def snapshot_state(b_list: List[float], u_list: List[str]) -> NDArray[Any]:
+    r"""
+    Implement a single snapshot state reconstruction.
+
+    Args:
+        b_list: The list of classical outcomes for the snapshot. Here,
+            b = 1 corresponds to :math:`|0\rangle`, and
+            b = -1 corresponds to :math:`|1\rangle`.
+
+        u_list: Array of ("X", "Y", "Z") for the applied
+            Pauli measurement on each qubit.
+
+    Returns:
+        Reconstructed snapshot in terms of nparray.
```

Comment:
```suggestion
    Returns:
        Reconstructed snapshot state as density matrix.
```

Reply:
actually, they can't be called density matrices, since "density matrices" has to be an element of $B(H)_*^+$ but it is not. You can think the physics reason that it is not a density matrix by it is not obtained by manipulating it in the lab/ by CPTP maps.  It is just a matrix.

Reply:
Ah okay, thanks for the explanation! Let's just say "matrix" then.

---

I take $B(H)_\ast^+$ to mean bounded linear operators on a Hilbert space $H$ that are positive definite ($+$), but what does the $\ast$ mean?

Reply:
oh, sorry for the jargon on the notations. You can understand it as kind of a dual space of B(H), 
this notation means predual, or trace class operators, where the definition could be found here: https://en.wikipedia.org/wiki/Trace_class#As_the_predual_of_bounded_operators